modules/custom/social_ajax_comments/social_ajax_comments.module

```diff
@@ -0,0 +1,138 @@
+        }
+        if (isset($build[$field_name])) {
+          $build['#cache']['contexts'][] = 'url';
+          $build[$field_name]['#attributes']['id'] = Html::getId($html_id);
+          Utility::setEntityRenderArray($build, $context['entity'], $context['display']->getMode());
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_entity_view_mode_alter().
+ */
+function social_ajax_comments_entity_view_mode_alter(&$view_mode, Drupal\Core\Entity\EntityInterface $entity, $context) {
+  // We need this to alter the view modes for Nodes as ajax_comments
+  // doesn't handle this correctly. And will add it as activity.
+  /** @var \Drupal\ajax_comments\FieldSettingsHelper $field_settings_helper */
+  $field_settings_helper = \Drupal::service('ajax_comments.field_settings_helper');
+
+  // We lookup through the node fields
+  // and check out that the attached file was provided.
+  $nodeFields = \Drupal::service('entity_field.manager')->getFieldDefinitions($entity->getEntityTypeId(), $entity->bundle());
+  foreach ($nodeFields as $nodeField) {
+    if ($nodeField->getType() === 'comment') {
+      // Load the configuration entity for the entity's view display settings.
+      /** @var \Drupal\Core\Entity\Display\EntityDisplayInterface $view_display */
+      $view_display = \Drupal::service('entity_type.manager')
+        ->getStorage('entity_view_display')
+        ->load($entity->getEntityTypeId() . '.' . $entity->bundle() . '.' . $view_mode);
+
+      if (empty($view_display)) {
+        continue;
+      }
+      $comment_formatter = $field_settings_helper->getFieldFormatter($view_display, $nodeField->getName(), $nodeField, $view_mode);
+
+      if (!empty($comment_formatter) && $field_settings_helper->isEnabled($comment_formatter)) {
+        /** @var \Drupal\ajax_comments\TempStore $tempStore */
+        $tempStore = \Drupal::service('ajax_comments.temp_store');
+        $route_match = \Drupal::routeMatch();
+        if ($entity instanceof Node && $route_match->getRouteName() === 'entity.node.canonical') {
+          $view_mode = 'default';
+        }
+        $tempStore->setViewMode($entity->getEntityType()->getLabel(), $view_mode);
+      }
+    }
+  }
+
+}
+
+/**
+ * Implements hook_comment_links_alter().
+ *
+ * Alter the links of a comment.
+ */
+function social_ajax_comments_comment_links_alter(array &$links, CommentInterface $entity, array &$context) {
+  $bundle = $entity->bundle();
+  $commented_entity = $entity->getCommentedEntity();
+  $account = \Drupal::currentUser();
+
+  if ($entity->isPublished() && $account->hasPermission('administer comments')) {
+    // For now we don't support edit ajax actions nor unpublishing.
+    $links['comment']['#links']['comment-edit']['attributes']['class'] = [];
+    $links['comment']['#links']['comment-edit']['url'] = Url::fromRoute('entity.comment.edit_form', ['comment' => $entity->id()]);
```

Comment:
Did you test whether simply adding the `Url::fromRoute()->toRenderable` already included the access checks that you need? Then perhaps the `$entity->isPublished` and `hasPermission` check are taken care of for you :)

See the implementation of the "Add content" links for an example.

Reply:
Yes, unfortunately we need the custom checks to override what Ajax comments does for us. Mainly because of UX flaws in the edit form that opens up. (Out of scope for now) Will add this in the code doc for explanation purposes. Good question.